python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -614,129 +615,124 @@ def null() -> "NullMetadataValue":
 
 
 @whitelist_for_serdes(storage_name="TextMetadataEntryData")
-class TextMetadataValue(
-    NamedTuple(
-        "_TextMetadataValue",
-        [
-            ("text", PublicAttr[Optional[str]]),
-        ],
-    ),
-    MetadataValue[str],
-):
+class TextMetadataValue(DagsterModel, MetadataValue[str]):
     """Container class for text metadata entry data.
 
     Args:
         text (Optional[str]): The text data.
     """
 
-    def __new__(cls, text: Optional[str]):
-        return super(TextMetadataValue, cls).__new__(
-            cls, check.opt_str_param(text, "text", default="")
-        )
+    text_inner: Optional[str] = Field(..., alias="text")
```

Comment:
[1] this object property (and `@property`s) could be `str` since we coerce `None` to `""` in the `__init__` that takes `Optional[str]`